### PR TITLE
Trajectory pt implementations

### PR DESCRIPTION
**Pull request by shaun-edwards**
This PR is dependent on #14.  It will capture the remaining items in PR #13.

### CODE REVIEW

**Comment by shaun-edwards** [comment]
@dpsolomon, I'm manually porting your changes in the PR.  It's not done yet, but I wanted to make sure everyone knew I was working on it.

**Comment by shaun-edwards** [comment]
This PR is ready for review.  It captures the remaining issues in PR #13.  It also lays the groundwork for a package rename #12 (namespace names corrected in a backwards compatible way).  Test methods were implemented to ensure that cartesian/joint points can be constructed.

One open issue (to be added) is the sampling of a waypoints and whether the waypoints or an external sampler should encapsulate that functionality.

@jrgnicho, please review.

File: descartes_trajectory_planning/include/descartes_trajectory_planning/trajectory_pt.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ struct Frame
   Frame(){};
   Frame(const Eigen::Affine3d &a):
     frame(a), frame_inv(a.inverse()) {};
+  Frame(const Frame &a):
```
Comment (jrgnicho) [inline descartes_trajectory_planning/include/descartes_trajectory_planning/trajectory_pt.h]: This copy constructor is equivalent to the default copy constructor provided by the compiler [here](http://www.cplusplus.com/articles/y8hv0pDG/).

**Comment by jrgnicho** [comment]
Just that one minor observation.  Other than that looks good to me.
+1

**Pull request merged**